src/gasman.h

```diff
@@ -293,13 +298,59 @@ EXPORT_INLINE UInt SIZE_BAG_CONTENTS(const void *ptr) {
 **  the application  must inform {\Gasman}  that it  has changed  the bag, by
 **  calling 'CHANGED_BAG(old)' in the above example (see "CHANGED_BAG").
 */
+#ifdef HPCGAP
+EXPORT_INLINE PURE_FUNC int ReadCheck(Bag bag)
```

Comment:
I really think we need to add an explanation here as to why we declare this function as a `PURE_FUNC`, and why this is "legal". That explanation should be written in one place, and other places using PURE_FUNC (at least in this header) should then refer to that: "For more information about the use of PURE_FUNC here, please read the documentation comment for ReadCheck" or something like that.

Reply:
Good point. Will do that.

Reply:
I've actually removed the `PURE_FUNC` stuff. It does not seem to make any performance difference anymore.